### PR TITLE
Fix unlocking the database

### PR DESCRIPTION
**Pull request by varjolintu**
When locking the database from the popup lock icon the database hashed are not updated properly to the contect script.

Fixes https://github.com/keepassxreboot/keepassxc-browser/issues/305.
Fixes https://github.com/keepassxreboot/keepassxc/issues/2480.

Also modifies the login popup to a locked state after clicking the lock icon.

### CODE REVIEW

File: keepassxc-browser/background/event.js [hunk 1]
```diff
@@ -87,7 +87,7 @@ kpxcEvent.onShowNotification = function(callback, tab, message) {
 
 kpxcEvent.showStatus = function(configured, tab, callback) {
     let keyId = null;
-    if (configured) {
+    if (configured && keepass.databaseHash !== 'no-hash') {
```
Comment (droidmonkey) [inline keepassxc-browser/background/event.js]: I do not like "magic values". Where is this set? Wouldn't a blank databaseHash be sufficient and just as telling?

File: keepassxc-browser/background/keepass.js [hunk 2]
```diff
@@ -1062,3 +1048,23 @@ keepass.updatePopup = function(iconType) {
...
+            keepass.updatePopup(configured ? 'normal' : 'cross');
+
+            // Send message to content script
+            browser.tabs.query({ active: true, currentWindow: true }).then((tabs) => {
+                if (tabs.length) {
+                    browser.tabs.sendMessage(tabs[0].id, {
+                        action: 'check_database_hash',
+                        hash: {old: keepass.previousDatabaseHash, new: keepass.databaseHash}
+                    });
+                    keepass.previousDatabaseHash = keepass.databaseHash;
```
Comment (droidmonkey) [inline keepassxc-browser/background/keepass.js]: If we are setting the previous hash to the "new" hash, when is the "new" hash set? This function makes no sense to me.

Comment (varjolintu) [reply keepassxc-browser/background/keepass.js]: The new hash is set when getDatabaseHash() has a response. This is called quite often.

File: keepassxc-browser/background/event.js (hunk 1 above)
Comment (varjolintu) [reply keepassxc-browser/background/event.js]: Probably. This is originally from chromeIPass. Basically `no-hash` means the database is locked.

Comment (droidmonkey) [reply keepassxc-browser/background/event.js]: Yeah i kind of figured, we can look at it later if necessary

File: keepassxc-browser/background/keepass.js (hunk 2 above)
Comment (droidmonkey) [reply keepassxc-browser/background/keepass.js]: Wouldn't this be subject to race conditions then?

Comment (varjolintu) [reply keepassxc-browser/background/keepass.js]: No. Updating these values cannot happen at the same time. In this case the update function is called only when user is locking or unlocking the database from KeePassXC or from the extension popup. And updating the current database hash only happens when doing some operation (retrieving credentials at page load etc).

File: keepassxc-browser/background/event.js (hunk 1 above)
Comment (varjolintu) [reply keepassxc-browser/background/event.js]: I changed the "magic value" to an empty string.

**Pull request merged**